docs/pages/build-reference/variants.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Installing app variants on the same device
+---
+
+When creating [development, preview, and production builds](../eas-json.md#common-use-cases), it's common to want to install one of each build on your device at the same time. This allows you to do development work, preview the next version of your app, and run the production version all on the same device, without needing to uninstall and reinstall the app.
+
+In order to be able to have multiple instances of an app installed on your device, each instance must have a unique Application ID (Android) or Bundle Identifier (iOS).
+
+**If you have a bare project**, you can accomplish this using flavors (Android) and targets (iOS). You can then configure which flavor is used with the `gradleCommand` field on your build profile, and you can configure the target that is used with the `scheme` field for iOS. 
+
+**If you have a managed project**, this can be accomplished by using **app.config.js** and environment variables in **eas.json**.
+
+## Example: configuring development and production variants
+
+Let's say we wanted a development build and production build of our managed Expo project. Your **eas.json** might look like this:
+
+```json
+{
+  "build": {
+    "development": {
+      "developmentClient": true
+    },
+    "production": {}
+  }
+}
+```
+
+And your **app.json** might look like this:
+
+```json
+{
+  "expo": {
+    "name": "MyApp",
+    "slug": "my-app",
+    "ios": {
+      "bundleIdentifier": "com.myapp"
+    },
+    "android": {
+      "package": "com.myapp"
+    }
+  }
+}
+```
+
+Let's convert this to **app.config.js** so we can make it more dynamic:
+
+```json
+export default {
+  name: "MyApp",
+  slug: "my-app",
+  ios: {
+    bundleIdentifier: "com.myapp",
+  },
+  android: {
+    package: "com.myapp",
+  },
+};
+```
+
+Now let's switch out the iOS `bundleIdentifier` and Android `package` (which becomes the Application ID) based on the presence of an environment variable in **app.config.js**:
+
+```js
+const IS_PROD = process.env.APP_VARIANT === "production";
```

Comment:
```suggestion
const identifierMap = {
  default: 'com.myapp'
  production: 'com.myapp'
  staging: 'com.myapp.staging'
  development: 'com.myapp.dev'
}
const applicationIdentifier = identifierMap[process.env.APP_VARIANT ?? 'default']
```

development clients have some gotchas, so maybe it would be better to not use default values in example, also I think using production as default is safer choice.

Reply:
the reason i defaulted to development client is because this way people don't need to remember to run `APP_ENV=development expo start` when they are building their app

Reply:
I wouldn't recommend that setup, because if someone runs `expo publish` with production release channel then they can by mistake publish OTA update that is changing their backend url to localhost(or sth like that)

Reply:
sounds good. updated to invert it so production is the default if no APP_VARIANT is given. i left it as `IS_DEV` instead of an identifier map, because i think people can figure out how to go from two to three+ variants based off of just js knowledge with this example, and it keeps the example more simple